plugins/device/deviceutils/store-utils.go

```diff
@@ -42,8 +41,29 @@ func GetDevices(peerIds ...string) ([]deviceapi.Info, error) {
 	return devices, nil
 }
 
-// DeviceExist checks the given device existence
-func DeviceExist(reqDevice string, devices []deviceapi.Info) bool {
+// FetchDevices returns devices from peer object.
+func FetchDevices(peerInfo *peer.Peer) ([]deviceapi.Info, error) {
+
+	var deviceInfo []deviceapi.Info
+	if len(peerInfo.Metadata["_devices"]) > 0 {
+		if err := json.Unmarshal([]byte(peerInfo.Metadata["_devices"]), &deviceInfo); err != nil {
+			return nil, err
+		}
+	}
+	return deviceInfo, nil
+}
+
+// CheckIfDeviceExist checks whether any device is registered with the peer or not.
+func CheckIfDeviceExist(peerInfo *peer.Peer) bool {
```

Comment:
@prashanthpai This function is required since there is a possibility that the path where device is stored changes, so its better if we keep this check in-store utils instead of directly accessing path in the code.